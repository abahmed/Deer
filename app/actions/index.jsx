// Local libraries
import { ACTION_TYPES } from './types';

/*
*  All notes related actions including payloads, if any
*/

// fetchAllNotes - Retrieve notes from the store

export function fetchAllNotes() {
  return {
    type: ACTION_TYPES.FETCH_ALL_NOTES,
  };
}

// addNote - Add/Save a new note to the store
export function addNote(payload) {
  return {
    type: ACTION_TYPES.ADD_NOTE,
    payload: payload,
  };
}

// const ACTIONS = {
//   addNote: addNote,
//   fetchAllNotes: fetchAllNotes
// }

// export default ACTIONS;

