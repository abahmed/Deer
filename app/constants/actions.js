export const ACTIONS = {
  GET_NEXT_LANG: 'SET_NEXT_LANG',
  TOGGLE_FADE: 'TOGGLE_FADE',
  UPDATE_EDITOR_STATE: 'UPDATE_EDITOR_STATE',
  FETCH_ALL_NOTES: 'FETCH_ALL_NOTES',
  UPDATE_NOTE_LIST: 'UPDATE_NOTE_LIST',
  SET_ACTIVE_NOTE: 'SET_ACTIVE_NOTE',
  UPDATE_NEW_NOTE_STATE: 'UPDATE_NEW_NOTE_STATE',
  UPDATE_SAVE_NOTE_STATE: 'UPDATE_SAVE_NOTE_STATE',
  ADD_NOTE: 'ADD_NOTE'
}
