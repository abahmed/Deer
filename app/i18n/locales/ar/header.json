{
  "newNoteBtn": "جديد",
  "saveBtn": "حفظ",
  "deleteBtn": "حذف",
  "settingsBtn":  "اﻷعدادات"
}
