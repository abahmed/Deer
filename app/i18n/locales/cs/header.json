{
  "newNoteBtn": "Nová poznámka",
  "saveBtn": "Uložit",
  "deleteBtn": "Smazat",
  "settingsBtn":  "Nastavení",
  "aboutUsBtn": "O Nás"
}
