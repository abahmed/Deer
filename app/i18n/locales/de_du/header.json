{
    "newNoteBtn": "Neue Notiz",
    "saveBtn": "Speichern",
    "deleteBtn": "Löschen",
    "settingsBtn": "Einstellungen",
    "aboutUsBtn": "Über uns"
}