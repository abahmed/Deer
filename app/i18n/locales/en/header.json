{
  "newNoteBtn": "New Note",
  "saveBtn": "Save",
  "deleteBtn": "Delete",
  "settingsBtn": "Settings",
  "aboutUsBtn": "About us"
}
