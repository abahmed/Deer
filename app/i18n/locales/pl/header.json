{
  "newNoteBtn": "Nowa Notatka",
  "saveBtn": "Zapisz",
  "deleteBtn": "Usu≈Ñ",
  "settingsBtn": "Ustawienia",
  "aboutUsBtn": "O nas"
}
