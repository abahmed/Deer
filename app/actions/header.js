import { createAction } from 'redux-actions'
import { ACTIONS } from '../constants/actions'

// Used for updating state of adding new note button (e.g. enabled or disabled).
export const setNewNoteDisabled = createAction(ACTIONS.UPDATE_NEW_NOTE_STATE)

// Used for updating state of save button (e.g. enabled or disabled).
export const setSaveDisabled = createAction(ACTIONS.UPDATE_SAVE_NOTE_STATE)

// Used for updating state of deleteing note button (e.g. enabled or disabled).
export const setDeleteDisabled = createAction(ACTIONS.UPDATE_DELETE_NOTE_STATE)
