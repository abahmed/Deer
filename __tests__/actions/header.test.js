import configureStore from 'redux-mock-store'

// Actions to be tested
import * as headerActions from '../../app/actions/header'

const mockStore = configureStore()
const store = mockStore()

describe('header_actions', () => {
  beforeEach(() => {
    store.clearActions()
  })

  describe('setNewNoteDisabled', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(headerActions.setNewNoteDisabled(true))
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('setSaveDisabled', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(headerActions.setSaveDisabled(true))
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  describe('setDeleteDisabled', () => {
    test('Dispatches the correct action and payload', () => {
      store.dispatch(headerActions.setDeleteDisabled(true))
      expect(store.getActions()).toMatchSnapshot()
    })
  })
})
