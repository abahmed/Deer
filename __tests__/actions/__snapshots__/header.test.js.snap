// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header_actions setDeleteDisabled Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": true,
    "type": "UPDATE_DELETE_NOTE_STATE",
  },
]
`;

exports[`header_actions setNewNoteDisabled Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": true,
    "type": "UPDATE_NEW_NOTE_STATE",
  },
]
`;

exports[`header_actions setSaveDisabled Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": true,
    "type": "UPDATE_SAVE_NOTE_STATE",
  },
]
`;
